bothub/api/v2/repository/views.py

```diff
@@ -843,6 +843,61 @@ def word_suggestions(self, request, **kwargs):
 
         return Response({"suggestions": suggestions})
 
+    @action(
+        detail=True,
+        methods=["POST"],
+        url_name="repository-examples",
+        serializer_class=RepositoryExampleSerializer,
+    )
+    def examples(self, request, **kwargs):
+        try:
+            repository = get_object_or_404(
+                Repository, pk=self.request.data.get("repository")
+            )
+        except DjangoValidationError:  # pragma: no cover
+            raise PermissionDenied()
+
+        try:
+            repository_version = get_object_or_404(
+                RepositoryVersion, pk=self.request.data.get("repository_version")
```

Comment:
why do the same consultation twice? you could simplify this query using filters

Reply:
in that case it's needed an repository_version object to be added on get_or_create a intent